### PR TITLE
WIP Windows Support for Cookbook

### PR DESCRIPTION
**Pull request by Techcadia**
This adds support for Windows in the cookbook. This is a Work in Progress Pull Request it should not be merged until its ready. We keep finding edge cases and have been fixing them.

Would love input on this change.

*NOTE:* We currently don't use the PKI for our platform which means I have not added it to support windows.

### CODE REVIEW

**Comment by Techcadia** [comment]
This also has https://github.com/Icinga/chef-icinga2/pull/286 in it currently because its what we need to make it function

**Comment by vkhatri** [comment]
@Techcadia Reviewing PR.

**Comment by Techcadia** [comment]
Currently I am out of office for the comming week. Totally up for hearing input just don't expect a response this week.

- David

File: attributes/default.rb [hunk 1]
```diff
@@ -59,14 +71,20 @@
 # icinga2 resources data bag
 default['icinga2']['databag'] = 'icinga2'
 
-default['icinga2']['run_dir'] = '/var/run/icinga2'
+default['icinga2']['var_dir'] = if node['platform'] == 'windows'
+                                  'C:/ProgramData/icinga2/var'
```
Comment (vills) [inline attributes/default.rb]: is windows work with slash, but not backslash? :-)

Comment (Techcadia) [inline attributes/default.rb]: Its might not be the best choice but it works for me currently on Server 2012 R2

Maybe a
https://github.com/chef-cookbooks/windows/blob/master/libraries/windows_helper.rb#L32-L34

would be a better choice

File: examples/icinga2_server/recipes/client.rb [hunk 2]
```diff
@@ -45,7 +45,7 @@
        "--host #{server_host} || " \
        "(echo 'Icinga2 Server not running';exit 1)"
   not_if { ::File.exist?(master_crt) }
-  notifies :reload, 'service[icinga2]', :delayed
+  notifies platform?('windows') ? :restart : :reload, 'service[icinga2]', :delayed
```
Comment (vkhatri) [inline examples/icinga2_server/recipes/client.rb]: How about creating an attribute for service notify action?

```ruby
default['icinga2']['service_notify_action'] = platform?('windows') ? :restart : :reload
```

**Comment by vkhatri** [comment]
PR LGTM. Will merge it shortly.

**Pull request merged**